test/xpu_api/utilities/meta/meta.trans/meta.trans.other/result_of.pass.cpp

```diff
@@ -0,0 +1,186 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+#include "support/utils_invoke.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct S
+{
+    typedef short (*FreeFunc)(long);
+    operator FreeFunc() const;
+    double
+    operator()(char, int&);
+    double const&
+    operator()(char, int&) const;
+    double volatile&
+    operator()(char, int&) volatile;
+    double const volatile&
+    operator()(char, int&) const volatile;
+};
+
+struct SD : public S
+{
+};
+
+struct NotDerived
+{
+};
+
+template <class Tp>
+struct Voider
+{
+    typedef void type;
+};
+
+template <class T, class = void>
+struct HasType : dpl::false_type
+{
+};
+
+template <class T>
+struct HasType<T, typename Voider<typename T::type>::type> : dpl::true_type
+{
+};
+
+template <typename T, typename U>
+struct test_invoke_result;
+
+template <typename Fn, typename... Args, typename Ret>
+struct test_invoke_result<Fn(Args...), Ret>
+{
+    static void
+    call()
+    {
+        static_assert(dpl::is_invocable<Fn, Args...>::value);
+        static_assert(dpl::is_invocable_r<Ret, Fn, Args...>::value);
+        ASSERT_SAME_TYPE(Ret, typename dpl::invoke_result<Fn, Args...>::type);
```

Comment:
Additional checks are absent here in original test.

Reply:
Since it is going to be part of our test infrastructure, the main question is do we want `dpl::invoke_result_t` to be tested. And I guess we want to. Otherwise `dpl::invoke_result_t` is untested.
I saw similar tests for other traits where the original trait and `trait_t` are tested simultaneously as part of one test case.

Reply:
Fixed